Create baseline scripts for inference microbenchmarking

dli357 opened the pull request:
# Description

Adds some scripts for baseline inference benchmarking execution for llama2-70b.

# Tests

Tested by manually executing the scripts.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed.

wang2yn84 commented on MaxText/inference/scripts/llama2_70b-inference-microbenchmark.sh:
```diff
@@ -0,0 +1,42 @@
+    hardware=gpu \
+    async_checkpointing=false \
+    per_device_batch_size=160 \
+    inference_microbenchmark_prefill_lengths=1024  \
```
Hi Dennis, why do we need both max_prefill_predict_length and inference_microbenchmark_prefill_lengths?

dli357 replied:
From what I can see, `max_prefill_predict_length` is passed to MaxEngine, whereas `inference_microbenchmark_prefill_lengths` is used for running the microbenchmark itself, so both should be required.

jwyang-google reviewed (commented):
for mlperf, we already have a script here: https://github.com/AI-Hypercomputer/maxtext/blob/main/MaxText/inference_mlperf/gpu/benchmarks_llama2-70b-h100_8.sh

wang2yn84 commented on MaxText/inference/scripts/llama2_70b-inference-mlperf.sh:
```diff
@@ -0,0 +1,39 @@
+PREFILL_LENS_AND_PER_DEVICE_BATCH_SIZES=1024,160 \
+MAXENGINE_ARGS="base_output_directory=/tmp/maxtext/ model_name=llama2-70b tokenizer_path=/opt/maxtext/assets/tokenizer.llama2 load_parameters_path=$1 quantize_kvcache=True attention=dot_product scan_layers=false hardware=gpu async_checkpointing=false ici_fsdp_parallelism=1 ici_autoregressive_parallelism=1 ici_tensor_parallelism=-1 weight_dtype=bfloat16 use_ragged_attention=false attention=dot_pro ...
+./llama_offline_run.sh
+cd $PREV_PWD
```
Nit, last time I checked the offline accuracy and it seems off. Basically the request doesn't match with response. Can you double check if the issue is still there? It should not block this PR from merging, but worth take a look.

dli357 replied:
Will address this in a follow-up change if needed, thanks!

wang2yn84 reviewed (approved):
Thank you Dennis!

dli357 commented:
> for mlperf, we already have a script here: https://github.com/AI-Hypercomputer/maxtext/blob/main/MaxText/inference_mlperf/gpu/benchmarks_llama2-70b-h100_8.sh

Thanks for the pointer! I've gone ahead and deleted the existing mlperf benchmark script, and renamed+refactored the microbenchmark script to more closely reflect similar usage patterns as the existing mlperf script.
